test/unit/org/apache/cassandra/db/rows/UnfilteredSerializerTest.java

```diff
@@ -0,0 +1,135 @@
+        ByteBuffer largeRow = getSerializedRow(50);
+        assertThatIOException().isThrownBy(() ->
+                                           testRowBodySerDe(10, buf -> replaceRowContent(buf, largeRow))).withMessageMatching("EOF after \\d+ bytes out of 50");
+    }
+
+    public static void testRowBodySerDe(int cellSize, Function<ByteBuffer, ByteBuffer> transform) throws IOException
+    {
+        Random random = new Random();
+        ByteBuffer data1 = ByteBuffer.allocate(cellSize);
+        ByteBuffer data2 = ByteBuffer.allocate(cellSize);
+        random.nextBytes(data1.array());
+        random.nextBytes(data2.array());
+
+        Row.Builder builder = BTreeRow.sortedBuilder();
+        builder.newRow(Clustering.EMPTY);
+        builder.addCell(BufferCell.live(md.regularColumns().getSimple(0), TimeUnit.MILLISECONDS.toMicros(System.currentTimeMillis()), data1.duplicate()));
+        builder.addCell(BufferCell.live(md.regularColumns().getSimple(1), TimeUnit.MILLISECONDS.toMicros(System.currentTimeMillis()), data2.duplicate()));
+        Row writtenRow = builder.build();
+
+        try (DataOutputBuffer out = new DataOutputBuffer())
+        {
+            UnfilteredSerializer.serializer.serialize(writtenRow, new SerializationHelper(SerializationHeader.makeWithoutStats(md)), out, 0, MessagingService.current_version);
+            out.flush();
+            try (DataInputBuffer in = new DataInputBuffer(transform.apply(out.asNewBuffer()), false))
+            {
+                builder = BTreeRow.sortedBuilder();
+                DeserializationHelper helper = new DeserializationHelper(md, MessagingService.current_version, DeserializationHelper.Flag.LOCAL, ColumnFilter.all(md));
+                Unfiltered readRow = UnfilteredSerializer.serializer.deserialize(in, SerializationHeader.makeWithoutStats(md), helper, builder);
+                assertEquals(writtenRow, readRow);
+            }
+        }
+    }
+
+    private ByteBuffer getSerializedRow(int cellSize) throws IOException
+    {
+        AtomicReference<ByteBuffer> rowData = new AtomicReference<>();
+        testRowBodySerDe(cellSize, buf -> {
+            rowData.set(buf.duplicate());
+            return buf;
+        });
+        return rowData.get();
+    }
+
+    private static ByteBuffer replaceRowContent(ByteBuffer original, ByteBuffer replacement)
+    {
+        try (DataOutputBuffer out = new DataOutputBuffer();
+             DataInputBuffer origIn = new DataInputBuffer(original, true);
+             DataInputBuffer replIn = new DataInputBuffer(replacement, false))
+        {
+            out.writeByte(origIn.readByte()); // flag
+            out.writeUnsignedVInt(origIn.readUnsignedVInt()); // row size
+            replIn.readByte(); // skip flag
+            replIn.readUnsignedVInt(); // skip row size
+            out.write(replacement);
+            out.close();
+            return out.asNewBuffer();
+        }
```

Comment:
I think a test for some corner case is needed such as what I said when limit is 0？

Reply:
No need for this